File: packages/token-squeezer/test/fixtures/sample.ts
// A simple utility module
import { readFile } from 'fs/promises';
import { join } from 'path';

/**
 * Process a config file and return parsed values
 */
export async function loadConfig(path: string): Promise<Record<string, unknown>> {
  const content = await readFile(join(process.cwd(), path), 'utf-8');
  return JSON.parse(content);
}

function validateConfig(config: Record<string, unknown>): boolean {
  if (!config || typeof config !== 'object') return false;
  return 'name' in config && 'version' in config;
}

// Internal helper
function mergeDefaults(user: any, defaults: any): any {
  const result = { ...defaults };
  for (const key of Object.keys(user)) {
    result[key] = user[key];
  }  
  return result;
}

File: packages/token-squeezer/test/fixtures/sample.py
"""
A sample Python module for TokenSqueezer testing.
"""

import json
import os
from typing import Dict, Any, Optional


def load_config(path: str) -> Dict[str, Any]:
    """Load a JSON config file and return its contents."""
    with open(path, 'r') as f:
        return json.load(f)


class ConfigManager:
    """Manages application configuration."""
    
    def __init__(self, defaults: Optional[Dict[str, Any]] = None):
        self._config = defaults or {}
    
    def get(self, key: str, default: Any = None) -> Any:
        return self._config.get(key, default)
    
    def set(self, key: str, value: Any) -> None:
        self._config[key] = value
    
    def validate(self) -> bool:
        # Check required keys
        required = ['name', 'version']
        for key in required:
            if key not in self._config:
                return False
        return True


# Internal helper
def _merge(base: Dict, overrides: Dict) -> Dict:
    result = {**base}
    result.update(overrides)
    return result
